cmd/apprepository-controller/controller_test.go

```diff
@@ -979,12 +842,10 @@ func Test_newSyncJob(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if tt.config.UserAgentComment != "" {
-				defaultConfig.UserAgentComment = tt.config.UserAgentComment
-				defer func() { defaultConfig.UserAgentComment = "" }()
```

Comment:
It wasn't clear to me why you were modifying `defaultConfig` at all here, given that you pass `tt.config` into your function below. But either way, although it's an interesting use of `defer` to reset the shared state for your test functions, it's a much better idea (IMO) to not have any shared state that requires resetting if you can avoid it (eg. what I did here by having each test construct the `Config` for each test).

Reply:
I didn't know either what was the reason for having this `defer` here. I just supposed it was the proper way to reset the state and I didn't want to change more things in the previous PR.
However, with this change, it is much clearer now :)